lib/Analysis/Allocation.cpp

```diff
@@ -129,6 +130,16 @@ ScratchConfig getScratchConfigForCvt(RankedTensorType srcTy,
   unsigned contiguousShapeDim = scratchConfig.repShape[scratchConfig.order[0]];
   scratchConfig.inVec = std::min(scratchConfig.inVec, contiguousShapeDim);
   scratchConfig.outVec = std::min(scratchConfig.outVec, contiguousShapeDim);
+  // Clamp the vector length to 128 / element bitwidth as this is the max
```

Comment:
The number `128` is a bit magic here. Is it possible to use enforce `128` in the backend, like getting it from targetInfo?

Reply:
agree, although the largest loads and stores from shmem in both nvidia and amd are of size 128. You still want me to push it to the backend or is it fine if I just make it a global constexpr next to the kPtrBitWidth?

Reply:
I remembered there's also a "128" in the pipelining path. Let's make it a constexpr for now

Reply:
done